src/main/tray/tray-menu-item/tray-menu-item-registrator.injectable.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { pipeline } from "@ogre-tools/fp";
+import { flatMap, kebabCase } from "lodash/fp";
+import type { Injectable } from "@ogre-tools/injectable";
+import { getInjectable } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import { extensionRegistratorInjectionToken } from "../../../extensions/extension-loader/extension-registrator-injection-token";
+import type { LensMainExtension } from "../../../extensions/lens-main-extension";
+import type { TrayMenuItem } from "./tray-menu-item-injection-token";
+import { trayMenuItemInjectionToken } from "./tray-menu-item-injection-token";
+import type { TrayMenuRegistration } from "../tray-menu-registration";
+
+const trayMenuItemRegistratorInjectable = getInjectable({
+  id: "tray-menu-item-registrator",
+
+  instantiate: (di) => (extension: LensMainExtension, installationCounter) => {
+    pipeline(
+      extension.trayMenus,
+
+      flatMap(toItemInjectablesFor(extension, installationCounter)),
```

Comment:
`installationCounter` has to be used till we migrate to `computedInjectMany` since it is used to handle the situation where extension is enabled, disabled and then re-enabled.

`computedInjectMany` is something that we already have, but we'll create separate PR for changing all of these to that format after auto-update work has been done.

Reply:
> since they need to be reacted to differently

No, our code should be made in a same way that the extensions do. That gives us simplicity and extendability. Having two separate sources is against that and therefore not preferrable.

Reply:
Then we shouldn't have any need for `injectMany` since everything can be pushed into a `IComputedValue<T>`... and we don't need to push so much into the injection library. Hiding the complexity doesn't seem to be a good idea either.

Reply:
What about all other benefits that `injectMany` gives us? Why would we want to give up those? E.g. repeatable solution for complying with Open Closed Principle, extendability. 

> Hiding the complexity doesn't seem to be a good idea either.

We want to get rid of the complexity for good. That happens by creating reusable solutions that have clear responsibility and utilizing them. E.g. in this case `injectMany` is the clear winner for making OpenLens code cleaner and easier to control. `injectMany` also complies with mindset that "everything is built as an extension" which leads to greater good.

Current mindset with extensions are that in Extension API 2.0 they will just introduce implementations for injection tokens (`injectMany`) which get registered when extension is enabled and de-registered when extension is disabled.

This mindset will make our Extension API very clean and all of the code handling extensions in OpenLens can be simplified to two functions which have clear responsibilities. (enableExtension -> register extension and  implementations, disableExtension -> de-register extension and all of its child).

Remember that the `registrators` are just adapters to allow us to refactor OpenLens code to place where everything supports `injectMany` mindset.

Reply:
Resolved for being trivial.

Reply:
Resolved for conversation to be had elsewhere. Note: behind this, there are a lot of long-spanning design pursuits, which are not at all apparent.